res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1411 @@
+        midi: [0x90+channelOffset, 0x47],
+        key: "play_stutter"
+    });
+    this.cue = new components.CueButton({
+        midi: [0x90 + channelOffset, 0x0C]
+    });
+    this.cuerewind = new components.Button({
+        midi: [0x90 + channelOffset, 0x48],
+        key: "start_stop"
+    });
+    this.jog = new components.JogWheelBasic({ // Jog Wheel is crap on the SR2, takes a big swing around to even get one midi tick. Use Platter instead.
+        deck: deckNumber,
+        wheelResolution: DDJSR2.scratchSettings.wheelResolution,
+        alpha: DDJSR2.scratchSettings.alpha,
+        beta: DDJSR2.scratchSettings.beta,
+        vinylMode: false, // Set the default to match the SR2's default.
+        inValueScale: function(value) {
+            if (!this.vinylMode) {
+                const delta = DDJSR2.getJogWheelDelta(value);
+                const newvalue = delta / 5 * DDJSR2.jogwheelSensitivity;
+                return newvalue;
+            } else {
+                return value < 0x40 ? value - (this.max + 1) : value;
+            }
+        }
+    });
+    this.tempoFader = new components.Pot({
+        invert: true,
+        key: "rate"
+    });
+    const rates = new DDJSR2.RingBufferView(DDJSR2.rateRanges);
+    this.tempoRange = new components.Button({
+        inKey: "rateRange",
+        input: function(channel, control, value, status, _group) {
+            if (this.isPress(channel, control, value, status)) {
+                this.inSetValue(rates.next());
+            }
+        }
+    });
+    this.tempoReset = new components.Button({
+        inKey: "rate",
+        input: function(channel, control, value, status, _group) {
+            if (this.isPress(channel, control, value, status)) {
+                this.inSetValue(0);
+            }
+        }
+    });
+    this.keyLock = new components.Button({
+        midi: [0x90 + channelOffset, 0x1A],
+        key: "keylock",
+        type: components.Button.prototype.types.toggle
+    });
+    this.needleSearchStripPositionShifted = new components.Pot({
+        group: theDeck.group,
+        key: "playposition"
+    });
+    this.needleSearchStripPosition = new components.Pot({
+        key: "playposition",
+        group: theDeck.group,
+        input: function(channel, control, value, _status, _group) {
+            if (!engine.getParameter(this.group, "play")) {
+                const newValue = this.inValueScale(value);
+                this.inSetParameter(newValue);
+            }
+        }
+    });
+    this.sync = new components.Button({
+        midi: [0x90 + channelOffset, 0x58],
+        key: "sync_enabled",
+        type: components.Button.prototype.types.toggle
+    });
+    this.syncOff = new components.Button({
+        midi: [0x90 + channelOffset, 0x5C],
+        key: "sync_enabled",
+        input: function(_channel, _control, _value, _status, _group) {
+            this.inSetValue(false);
+        }
+    });
```

Comment:
you'll probably don't want both controlling the same led.
```suggestion
    this.sync = new components.Button({
        midi: [0x90 + channelOffset, 0x58],
        key: "sync_enabled",
        type: components.Button.prototype.types.toggle
    });
    this.syncOff = new components.Button({
        midi: [0x90 + channelOffset, 0x5C],
        inKey: "sync_enabled",
        input: function(_channel, _control, _value, _status, _group) {
            this.inSetValue(false);
        }
    });
```